### PR TITLE
feat(datahub-gms): Enable autoscaling via HPA

### PR DESCRIPTION
**Pull request by 7onn**
## Summary
https://github.com/datahub-project/datahub/issues/11761
The company where I work for, started crashing datahub-gms during Snowflake ingestion, and I thought it would be handy to have an autoscaler for this workload. Hence, this pull request.

## How to test it?
Edit the values.yaml of both datahub chart, and datahub-gms subchart. Make sure to enable `datahub-gms.hpa.enabled` and `global.datahub_standalone_consumers_enabled`. Then, navige to the subchart folder and:
```
cd charts/datahub/subcharts/datahub-gms
helm template test . --values values.yaml --values ../../values.yaml | yq '. | select(.kind == "HorizontalPodAutoscaler")'
```

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: charts/datahub/values.yaml [hunk 1]
```diff
@@ -446,7 +446,7 @@ global:
   strict_mode: true
   graph_service_impl: elasticsearch
   datahub_analytics_enabled: true
-  datahub_standalone_consumers_enabled: false
+  datahub_standalone_consumers_enabled: true
```
Comment (7onn) [inline charts/datahub/values.yaml:449]:
```suggestion
  datahub_standalone_consumers_enabled: false
```

File: .github/workflows/lint-test.yaml [hunk 2]
```diff
@@ -1,9 +1,11 @@
 name: Lint and Test Charts
 
 on:
+  workflow_dispatch:
   push:
     branches:
       - master
+      - 7onn/autoscale-gms
```
Comment (7onn) [inline .github/workflows/lint-test.yaml:8]:
```suggestion
```

File: .github/workflows/lint-test.yaml [hunk 3]
```diff
@@ -1,9 +1,11 @@
 name: Lint and Test Charts
 
 on:
+  workflow_dispatch:
```
Comment (7onn) [inline .github/workflows/lint-test.yaml:4]:
```suggestion
```

**Comment by david-leifker** [comment]
This looks good to me. I would be interested to know how this works in production, can GMS scale quick enough to help with load spikes. I can see this being helpful for long running ingestion runs for sure.

Thank you!

**Review by david-leifker — APPROVED** [review]

**Comment by 7onn** [comment]
> This looks good to me. I would be interested to know how this works in production, can GMS scale quick enough to help with load spikes. I can see this being helpful for long running ingestion runs for sure.
> 
> Thank you!

Hi David, first of all, thanks a lot for your time reviewing my contribution. I highly appreciate it.

In regards to production, I suppose the default setting isn't the most helpful thing with `targetCPUUtilizationPercentage: 100` as this would require the app to be under some spike for some time and perhaps even get throttled for a while which could cause the health check to fail and the pod to restart. What I had planned was to release this with `targetCPUUtilizationPercentage: 60` to scale out precociously and not even let GMS get throttled by Kubernetes.

**Pull request merged**